clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -156,7 +177,8 @@ else if (LZ4.name.equals(name))
     private static class LZ4Constructors {
         static final MethodHandle INPUT = findConstructor(
                 "org.apache.kafka.common.record.KafkaLZ4BlockInputStream",
```

Comment:
ok, I've change it to construct the lz4 streams directly, and updated the comments accordingly.